src/OrchardCore/OrchardCore.DisplayManagement/Layout/LayoutShapes.cs

```diff
@@ -22,6 +34,21 @@ public void Discover(ShapeTableBuilder builder)
                     layout.Tail = await created.ShapeFactory.CreateAsync("DocumentZone", new { ZoneName = "Tail" });
 
                     layout.Content = await created.ShapeFactory.CreateAsync("Zone", new { ZoneName = "Content" });
+
+                    var httpContext = _accessor.HttpContext;
+                    if (httpContext != null)
+                    {
+                        if (AdminAttribute.IsApplied(httpContext))
+                        {
+                            layout.Metadata.Alternates.Add("Layout__Admin");
```

Comment:
What's the point if the Admin has its own theme, so the Layout file of an admin theme is the one what will be picked. Otherwise give me an example where `Layout-Admin.cshtml` would be used instead of `Layout.cshtml`.

Reply:
We could do without this, I suppose. The reasoning for it, is that if you selected a theme and it has no layout, it should generate an error (in my opinion) instead of using the Layout from the admin. I thought it would be confusing to the developer, an error is more important explicit.